### PR TITLE
Fixes/cleans prod db cleansing task

### PR DESCRIPTION
**Pull request by ivar**
I was having issues cleaning the prod db. These are the fixes I made to get it working. 
This is extracted from another working branch on my machine, but I figured it'd be better to review separately.

### CODE REVIEW

File: lib/tasks/production-copy.rake [hunk 1]
```diff
@@ -44,12 +44,10 @@ namespace :production_copy do
   end
 
   desc "Clean up BalancedPayments refs and user email and passwords"
-  task :cleanse do
+  task cleanse: [:environment] do
```
Comment (ivar) [inline lib/tasks/production-copy.rake]: If you call the cleanse task by itself it'll fail without this change.

File: lib/tasks/production-copy.rake [hunk 2]
```diff
@@ -264,23 +262,23 @@ module CloneProductionHelper
 
   def balanced_payments_refs_to_clear
     if ENV["KEEP_BALANCED_URIS"] =~ /^(y|on|true)/i
-      [ { model: Payment,     fields: [ :balanced_uri ] } ]
+      [ { model: ::Payment,     fields: [ :balanced_uri ] } ]
```
Comment (ivar) [inline lib/tasks/production-copy.rake]: ensures proper namespacing.. was getting error _"Could not find CloneProductionHelper::Payment"_

File: lib/tasks/production-copy.rake [hunk 3]
```diff
@@ -22,7 +22,7 @@ namespace :production_copy do
   end
 
   desc "Get a local copy of production (no cleansing)"
-  task fetch_db: [:environment]  do
+  task fetch_db: [:environment] do
```
Comment (thermistor) [inline lib/tasks/production-copy.rake]: You don't need an array, just use:

    task fetch_db: :environment do

and below for the `cleanse` task as well.

**Comment by ivar** [comment]
@thermistor PR changes made - ready for re-review

File: lib/tasks/production-copy.rake [hunk 4]
```diff
@@ -306,13 +304,17 @@ module CloneProductionHelper
...
-      unless email =~ /@example\.com$/ or email =~ /atomicobject/ or email =~ /localorb/
+      if email.present? && email_needs_cleansing?(email)
         new_email = user.email.gsub("@","_at_") + "@example.com"
         user.update_columns(email: new_email)
       end
     end
   end
 
+  def email_needs_cleansing?(email)
+    !(email =~ /@example\.com$/ || email =~ /atomicobject/ || email =~ /local||b/)
```
[the remark this answers is not in the document]
Comment (ivar) [reply lib/tasks/production-copy.rake]: This improvement is on purpose. Using ruby's keyword `or` is [generally frowned upon](http://www.virtuouscode.com/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/) (except for control flow)as it's rank in the operator order of precedence is really low and very confusing.

[the remark this answers is not in the document]
Comment (rbarreca) [reply lib/tasks/production-copy.rake]: But did you mean to change `localorbit` to `local||bit`?

File: lib/tasks/production-copy.rake [hunk 5]
```diff
@@ -286,7 +284,7 @@ module CloneProductionHelper
     refs_to_clear += balanced_payments_refs_to_clear
     refs_to_clear += stripe_refs_to_clear
 
-    refs_to_clear.each do |model:,fields:|
+    refs_to_clear.each do |model, fields|
```
Comment (ivar) [inline lib/tasks/production-copy.rake]: this was a weird thing. I'm not sure how it worked before..

File: lib/tasks/production-copy.rake (hunk 4 above)
[the remark this answers is not in the document]
Comment (ivar) [reply lib/tasks/production-copy.rake]: whoops. nice catch!

[the remark this answers is not in the document]
Comment (ivar) [reply lib/tasks/production-copy.rake]: fixed

File: lib/tasks/production-copy.rake (hunk 5 above)
Comment (ivar) [reply lib/tasks/production-copy.rake]: confirmed that this fails.. I experimented with changing the names of the arguments to the block, I couldn't find a way to access said arguments.

Comment (rbarreca) [reply lib/tasks/production-copy.rake]: @ivar @thermistor removal of colons has broken `cleanse` for me (opposite of what Ivar found). Is this a difference of environment or was it a mistake while testing? I opened issue #3457 for this.

**Pull request merged**